scripts/launch/connect-to-cluster/default.nix

```diff
@@ -28,37 +32,58 @@ let
       fallbacks: 7
   '';
+  configFiles = pkgs.runCommand "cardano-config" {} ''
+    mkdir -pv $out
+    cd $out
+    cp -vi ${iohkPkgs.cardano-sl.src + "/configuration.yaml"} configuration.yaml
+    cp -vi ${iohkPkgs.cardano-sl.src + "/mainnet-genesis-dryrun-with-stakeholders.json"} mainnet-genesis-dryrun-with-stakeholders.json
+    cp -vi ${iohkPkgs.cardano-sl.src + "/mainnet-genesis.json"} mainnet-genesis.json
+    cp -vi ${iohkPkgs.cardano-sl.src + "/../scripts/log-templates/log-config-qa.yaml"} log-config-qa.yaml
+    cp -vi ${if topologyFile != null then topologyFile else topologyFileDefault } topology.yaml
+  '';
 in pkgs.writeScript "${executable}-connect-to-${environment}" ''
+  #!${pkgs.stdenv.shell}
+
   if [[ "$1" == "--delete-state" ]]; then
     echo "Deleting ${stateDir} ... "
-    rm -Rf ${stateDir}               
+    rm -Rf ${stateDir}
   fi
 
   echo "Keeping state in ${stateDir}"
   mkdir -p ${stateDir}/logs
 
-  echo "Launching a single node connected to '${environment}' ..."
+  echo "Launching a node connected to '${environment}' ..."
+  ${ifWallet ''
+  if [ ! -d ${stateDir}tls ]; then
+    mkdir ${stateDir}/tls/
+    openssl req -x509 -newkey rsa:2048 -keyout ${stateDir}/tls/server.key -out ${stateDir}/tls/server.cert -days 3650 -nodes -subj "/CN=localhost"
+  fi
+  ''}
+
 
   ${executables.${executable}}                                     \
-    ${ ifWallet "--web"}                                           \
     --no-ntp                                                       \
-    --configuration-file ${src}/node/configuration.yaml            \
+    --configuration-file ${configFiles}/configuration.yaml         \
     --configuration-key ${environments.${environment}.confKey}     \
-    ${ ifWallet "--tlscert ${src}/scripts/tls-files/server.crt"}   \
-    ${ ifWallet "--tlskey ${src}/scripts/tls-files/server.key"}    \
-    ${ ifWallet "--tlsca ${src}/scripts/tls-files/ca.crt"}         \
-    --log-config ${src}/scripts/log-templates/log-config-qa.yaml   \
-    --topology "${topologyFile}"                                   \
+    ${ ifWallet "--web"}                                           \
+    ${ ifWallet "--tlscert ${stateDir}/tls/server.cert"}           \
+    ${ ifWallet "--tlskey ${stateDir}/tls/server.key"}             \
+    ${ ifWallet "--tlsca ${stateDir}/tls/server.cert"}             \
+    --log-config ${configFiles}/log-config-qa.yaml                 \
+    --topology "${configFiles}/topology.yaml"                      \
     --logs-prefix "${stateDir}/logs"                               \
     --db-path "${stateDir}/db"                                     \
     ${ ifWallet "--wallet-db-path '${stateDir}/wallet-db'"}        \
-    --keyfile ${stateDir}/secret.key                   
+    --keyfile ${stateDir}/secret.key                               \
+        ${ ifWallet "--wallet-address ${walletListen}"}    \
+        --ekg-server 127.0.0.1:8080 --metrics +RTS -T -RTS             
```

Comment:
docker cant forward things to `127.0.0.1`, thats why the wallet has to listen on `0.0.0.0` when docker is in use

Reply:
Tracked in CSL-2139